### PR TITLE
stalonetray: 0.8.3 -> 0.8.4

### PR DESCRIPTION
**Pull request by dschrempf**
This update is quite involved for a minor version update.

1. I had to change the source from Sourceforge to Github.

2. I manually specify the autoconf pre-configuration steps. This seems to be
necessary, but is there a better, more canonical way?

3. I had to add the docbook dependency and tinker around with substitutions,
because the original files would pull sources from an online repository. Please
have a look!

Thanks for your comments.

Cheers.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/stalonetray/default.nix [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
+        autoconf
+        automake --add-missing
+      '';
 
   hardeningDisable = [ "format" ];
 
   meta = with lib; {
     description = "Stand alone tray";
-    homepage = "http://stalonetray.sourceforge.net";
+    homepage = "https://github.com/kolbusa/stalonetray";
```
Comment (7c6f434c) [inline pkgs/applications/window-managers/stalonetray/default.nix:58]: (just so this is recorded) checked the upstream redirect

File: pkgs/applications/window-managers/stalonetray/default.nix [hunk 2]
```diff
@@ -1,29 +1,63 @@
...
+      dtd_str_sub = "${docbook_xml_dtd_44}/xml/dtd/docbook/docbookx.dtd";
+    in
+      ''
+        substituteInPlace configure.ac --replace '${ac_str}' '${ac_str_sub}'
+        substituteInPlace stalonetray.xml.in --replace '${dtd_str}' '${dtd_str_sub}'
+        autoupdate
+        aclocal
+        autoheader
+        autoconf
+        automake --add-missing
```
Comment (7c6f434c) [inline pkgs/applications/window-managers/stalonetray/default.nix:51]: I think the canonical way to get this done in the correct order is adding `autoreconfHook` to `nativeBuildInputs`.

File: pkgs/applications/window-managers/stalonetray/default.nix [hunk 3]
```diff
@@ -1,29 +1,63 @@
...
+    let
+      db_root = "${docbook-xsl-ns}/share/xml/docbook-xsl-ns";
+      ac_str = "AC_SUBST(DOCBOOK_ROOT)";
+      ac_str_sub = "DOCBOOK_ROOT=${db_root}; ${ac_str}";
+      dtd_str = "http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd";
+      dtd_str_sub = "${docbook_xml_dtd_44}/xml/dtd/docbook/docbookx.dtd";
+    in
+      ''
+        substituteInPlace configure.ac --replace '${ac_str}' '${ac_str_sub}'
+        substituteInPlace stalonetray.xml.in --replace '${dtd_str}' '${dtd_str_sub}'
```
Comment (7c6f434c) [inline pkgs/applications/window-managers/stalonetray/default.nix:46]: Given that `xsltproc` gets `--nonet`, should the first part suffice?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136188` at 11345ae6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>stalonetray</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchurl`.
Near pkgs/applications/window-managers/stalonetray/default.nix:6:3:
```
  |
6 | , fetchurl
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/window-managers/stalonetray/default.nix:59:5:
```
   |
59 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `xorgproto`.
Near pkgs/applications/window-managers/stalonetray/default.nix:12:3:
```
   |
12 | , xorgproto
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/window-managers/stalonetray/default.nix (hunk 3 above)
Comment (dschrempf) [reply pkgs/applications/window-managers/stalonetray/default.nix]: Thank you, the first line is indeed enough if the docbook is provided separately.

**Comment by dschrempf** [comment]
I also implemented all the suggestions from nixpkgs-review.

**Pull request merged**